### PR TITLE
CN-ViewAPostsComments#19

### PR DESCRIPTION
**Pull request by CaraNetzer**
# Description

[Trello Ticket #19](https://trello.com/c/9QVUzOLr)

Added functionality to view a list of comments associated with a post from its details page. If a post does not have any comments yet, the View Comments button will not be displayed on the post's details page.

# Testing Instructions

1. `git add` and `git commit` whatever you're working on on your branch
2. `git checkout main`
3. `git fetch --a`
4. `git checkout CN-ViewAPostsComments#19`
5. If you don't see the most recent version of my files, `git pull origin CN-ViewAPostsComments#19`
6. If you haven't created the Tabloid MVC database, please do so.
7. Use this data to fill out tables and columns: [SQL Tables](https://github.com/NewForce-Cohort-6/tabloidmvc-the-tootsie-rolls/blob/main/SQL/01_TabloidMVC_Create_DB.sql)
8. Use this to seed the database : [SQL SEED DATA](https://github.com/NewForce-Cohort-6/tabloidmvc-the-tootsie-rolls/blob/main/SQL/02_TabloidMVC_Seed_Data.sql)
9. Next seed some comment data:

```
--SET IDENTITY_INSERT Comment ON
--INSERT INTO Comment (PostId, UserProfileId, [Subject], Content, CreateDateTime)
--VALUES (1, 1, 'test1','comment 1',05/29/2015), 
--	(1, 1, 'test2', 'comment 2', 10/31/2022), 
--	(1, 1, 'test3', 'comment 3', 10/30/2022),
--	(1, 1, 'test4', 'comment 4', 10/29/2022),
--  (2, 1, 'test5', 'comment 5', 10/29/2022);
--SET IDENTITY_INSERT Comment OFF
```
10. Run `select * from comment;` to make sure all the comments are successfully in the datebase
11. If the dates do not seed correctly, run the following commands with custom dates for each comment:
```
update comment
set 
	CreateDateTime = '2022-08-05 00:00:00.000'
where id = 5;

update comment
set 
	CreateDateTime = '2022-03-05 00:00:00.000'
where id = 4;

update comment
set 
	CreateDateTime = '2022-01-05 00:00:00.000'
where id = 3;

update comment
set 
	CreateDateTime = '2022-06-05 00:00:00.000'
where id = 2;

update comment
set 
	CreateDateTime = '2022-02-05 00:00:00.000'
where id = 1;
```

12. Click the execute with debugger solid green button to launch the app
13. Log into the website using “[admin@example.com](mailto:admin@example.com)” if greeted with a login page
14. From the home page, click the “POST” option in the navbar
15. If you do not have at least three posts in your database, add posts until you have at least three. `asdfasdfdsfd` is fine for most/all of the fields. In our database we have four comments associated with the first post, and one associated with a second. So we want both of these posts available to view, as well as one that has no comments.
16. Click on the Eye icon to view the Details page of the first post. A View Comments button should be visible at the bottom of the screen next to the Back to Posts button.
18. Select the View Comments button. You should be directed to the Comments list page for the Post
19. There should be four comments listed in order of creation date with the most recent on top (In this case the order should be test2, test4, test1, test3). 
20. The `Title` of the related Post should be displayed at the top of the page (C# is the Best Language) as well as `Subject` `Content` `Author's Display Name` `Creation date (MM/DD/YYYY)` in the table
21. Click Back to Posts
22. Click on the Eye icon to view the Details page of the second post. A View Comments button should be visible at the bottom of the screen next to the Back to Posts button.
23. Select the View Comments button. You should be directed to the Comments list page for the Post
24. There should be one comment listed (test5) with the Title you assigned it at the top of the page
25. Click Back to Posts
22. Click on the Eye icon to view the Details page of the third post
23. There should be no View Comments button available

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings or errors
- [x] I have added test instructions that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by sarahebrooks12** [comment]
Testing instructions look good --- get 2 testing approvals before merging 👼

**Review by jacobjjude — APPROVED** [review]
🌟 Pulled down, tested. Works as intended.

Only issue I ran into is that my comments were not in the order listed in instructions, but were listed in order from most recent to least, so my seed data must have been off. I don't think this affects functionality.

**Review by robert-brunner — APPROVED** [review]
Your seed data is a little caddywampus.  
it should actually be-  INSERT INTO Comment (PostId, UserProfileId, [Subject], Content, CreateDateTime)
VALUES (1, 1, 'test1','comment 1',05/29/2015), 
	(1, 1, 'test2', 'comment 2', 10/31/2022), 
	(1, 1, 'test3', 'comment 3', 10/30/2022),
	(1, 1, 'test4', 'comment 4', 10/29/2022),
  (2, 1, 'test5', 'comment 5', 10/29/2022);
  
  
  Code works great though.

**Pull request merged**
